fedimint-core/src/module/mod.rs

```diff
@@ -194,15 +216,17 @@ impl ApiEndpoint<()> {
         async fn handle_request<'a, 'b, E>(
             state: &'a E::State,
             dbtx: &mut fedimint_core::db::ModuleDatabaseTransaction<'b, ModuleInstanceId>,
-            param: E::Param,
+            request: ApiRequest<E::Param>,
+            api_auth: ApiAuth,
         ) -> Result<E::Response, ApiError>
         where
             E: TypedApiEndpoint,
             E::Param: Debug,
             E::Response: Debug,
         {
-            tracing::trace!(target: "fedimint_server::request", ?param, "recieved request");
-            let result = E::handle(state, dbtx, param).await;
+            tracing::trace!(target: "fedimint_server::request", ?request, "recieved request");
+            let has_auth = request.auth == Some(api_auth);
```

Comment:
So we are handing in `api_auth` from the config just for the comparison here? I'd really hope there is a better way to go about it in the future, but can be done in a separate PR since it would likely involve some more refactoring and isn't high prio imo.

Reply:
I think there might be a way to do it but would require moving the `ServerConfig` into `fedimint-core`.